### PR TITLE
feat: create email input pattern tckt-362

### PR DESCRIPTION
**Pull request by kalasgarov**
Created Email Input Pattern for Forms Users.
Created Email Input Pattern Edit for Form Builders.
Added unit tests for Email Input config file.

### CODE REVIEW

File: packages/forms/src/patterns/email-input/email-input.test.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+      const inputValue = { email: 'testEmail@test.com' };
+      if (!emailInputConfig.parseUserInput) {
+        expect.fail('emailInputConfig.parseUserInput is undefined');
+      }
+      const result = emailInputConfig.parseUserInput(pattern, inputValue);
+      if (result.success) {
+        expect(result.data).toEqual(inputValue);
+      } else {
+        throw new Error('Unexpected validation failure');
```
Comment (danielnaab) [inline packages/forms/src/patterns/email-input/email-input.test.ts:58]:
```suggestion
        expect.fail('Unexpected validation failure');
```

File: packages/forms/src/patterns/email-input/email-input.test.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+      const inputValue = { email: 'testEmail.co' };
+      if (!emailInputConfig.parseUserInput) {
+        expect.fail('emailInputConfig.parseUserInput is undefined');
+      }
+      const result = emailInputConfig.parseUserInput(pattern, inputValue);
+      if (!result.success) {
+        expect(result.error).toBeDefined();
+      } else {
+        throw new Error('Unexpected validation success');
```
Comment (danielnaab) [inline packages/forms/src/patterns/email-input/email-input.test.ts:80]:
```suggestion
        expect.fail('Unexpected validation success');
```

File: packages/forms/src/patterns/email-input/email-input.test.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        required: true,
+      };
+
+      const schema = createEmailSchema(data);
+      const validInput = { email: 'testEmail@test.com' };
+      const invalidInput = { email: 'testEmail.com' };
+
+      expect(schema.safeParse(validInput).success).toBe(true);
+      expect(schema.safeParse(invalidInput).success).toBe(false);
+    });
```
Comment (danielnaab) [inline packages/forms/src/patterns/email-input/email-input.test.ts:22]: Looks like Prettier could be run on this.

**Review by danielnaab — COMMENTED** [review]
I had a couple comments, but otherwise this looks great.

File: packages/forms/src/patterns/email-input/email-input.test.ts (hunk 1 above)
Comment (kalasgarov) [reply packages/forms/src/patterns/email-input/email-input.test.ts]: Updated.

File: packages/forms/src/patterns/email-input/email-input.test.ts (hunk 2 above)
Comment (kalasgarov) [reply packages/forms/src/patterns/email-input/email-input.test.ts]: Updated.

File: packages/forms/src/patterns/email-input/email-input.test.ts (hunk 3 above)
Comment (danielnaab) [reply packages/forms/src/patterns/email-input/email-input.test.ts:22]: Oh sorry, it's just a formatting issue on Github.

**Review by danielnaab — APPROVED** [review]

**Pull request merged**